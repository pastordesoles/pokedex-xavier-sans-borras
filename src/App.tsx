import React from "react";
import { useEffect } from "react";
import useApi from "./hooks/useApi";

const App = (): JSX.Element => {
  const { loadAllPokemon } = useApi();
  useEffect(() => {
    (async () => {
      loadAllPokemon();
    })();
  }, [loadAllPokemon]);
  return <h1>In progress</h1>;
};

export default App;
