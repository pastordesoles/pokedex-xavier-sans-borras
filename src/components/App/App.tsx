import { Route, Routes } from "react-router-dom";
import HomePage from "../../pages/HomePage/HomePage";

const App = (): JSX.Element => {
  return (
    <Routes>
      <Route path="/pokemon" element={<HomePage />} />
    </Routes>
  );
};

export default App;
