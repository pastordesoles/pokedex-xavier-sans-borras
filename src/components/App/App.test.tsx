import { MemoryRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import App from "./App";

describe("Given an App component", () => {
  describe("When it's rendered with path '/*'", () => {
    test("Then it should render 'not found page'", () => {
      const expectedApp = TestRenderer.create(
        <MemoryRouter initialEntries={["/*"]}>
          <App />
        </MemoryRouter>
      );

      expect(expectedApp).toMatchSnapshot();
    });
  });
});
