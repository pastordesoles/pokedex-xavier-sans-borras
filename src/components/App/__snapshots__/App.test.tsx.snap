// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When it's rendered with path '/*' Then it should render 'not found page' 1`] = `
<nav
  className="sc-gKPRtg gCPdIL header__navigation"
>
  <a
    className="header__link"
    href="/pokemon"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    className="header__link"
    href="/favourites"
    onClick={[Function]}
  >
    Favourites
  </a>
</nav>
`;
